### ISSUE TITLE
NaN or Inf found in input tensor.

### ISSUE THREAD

**Issue by ingra14m** [liuyuan-pal/NeRO#6]
When I tried to train materials after Stage I, nan or inf is found in the training process, causing the loss turn to be nan.

**Comment by liuyuan-pal** [comment]
Hi, which object are you training to get the NaN loss.

**Comment by ingra14m** [comment]
I found this problem with all objects. The outputs are shown below:
```shell
warn!!! the normals are flipped in NeuS by default. You may flip the normal according to your mesh!
| 9/100000 [00:03<6:45:53,  4.11it/s, loss=nan, lr=4e-6]NaN or Inf found in input tensor.
NaN or Inf found in input tensor.
NaN or Inf found in input tensor.
```
Since BRDF estimation depends on the mesh obtained in the first stage, is there any problem in the mesh I have extracted.

**Comment by liuyuan-pal** [comment]
You may visualize the extracted mesh using Meshlab to check for any problems in stage I. I've checked that the training is OK on my machine and does not have nan losses.

**Comment by ingra14m** [comment]
Tks for ur response. This is the extracted mesh from the original source code. It seems that the normal is flipped. I have no idea if there is anything wrong with this result.
![mesh](https://github.com/liuyuan-pal/NeRO/assets/63096187/001cb821-57f5-4d24-8580-79d690c833bf)

**Comment by liuyuan-pal** [comment]
It's OK to have these normals flipped. You may click the `double` at the right bottom to visualize the mesh. The mesh seems to be fine from the silhouette. You may check the nan starts from which line.
```
if torch.sum(torch.isnan(<some-tensors-here>))>0: import ipdb; ipdb.set_trace()
```
![image](https://github.com/liuyuan-pal/NeRO/assets/23974453/c5d61e26-1910-4e28-b63e-03e8c215aec3)

**Comment by ingra14m** [comment]
Tks for ur answer. Seems that the nan values first appear in the normal obtained by ray tracing.
![ray-trace](https://github.com/liuyuan-pal/NeRO/assets/63096187/cac948e8-a0fe-4b7f-b676-8b7ff5821ef0)

**Comment by liuyuan-pal** [comment]
https://github.com/liuyuan-pal/NeRO/blob/11084150944e926cb7fea3214e6a5743b9d679ef/raytracing/raytracer.py#L49

It's probably related to the installation of the ray tracer. You may check this line contains nan or not.

**Comment by ingra14m** [comment]
Thanks for ur response. I have already fix the bug. Thank u!

**Comment by gs14iitbbs** [comment]
> Thanks for ur response. I have already fix the bug. Thank u!

May I know how you solved?
